src/main/java/teammates/ui/controller/FeedbackSubmissionEditSaveAction.java

```diff
@@ -144,8 +144,9 @@ private void saveResponse(FeedbackResponseAttributes response)
             throws EntityDoesNotExistException {
         if (response.getId() != null) {
             // Delete away response if any empty fields
-            if (response.responseMetaData.getValue().isEmpty() || 
-                response.recipientEmail.isEmpty()) {
+            if (response.responseMetaData.getValue().isEmpty() ||
+                    response.recipientEmail.isEmpty() ||
+                    isEmptyContributionResponse(response)) {
```

Comment:
We should also take note that until #3139 is completed, any change to FeedbackSubmissionEditPageAction and FeedbackSubmissionEditSaveAction needs to be made in FeedbackQuestionSubmissionEditPageAction and FeedbackQuestionSubmissionEditSaveAction as well.